### ISSUE TITLE
San Francisco Map is still rotated

### ISSUE THREAD

**Issue by Linsha350** [lgsvl/simulator#1792]
Hi.

I Download lgsvl 2021.3 version.
And I found apollo 5.0 is updated with HD Map.
So I pulled lgsvl/apollo 5.0 with 105f7fd19220dc4c04be1e075b1a5d932eaa2f3f and did clean build.

The San Francisco Map file is changed.
However, The localization is still rotated.
Previous apollo 5.0 commit with lgsvl 2021.2 has same result this apollo 5.0 commit with lgsvl 2021.3.
![Screenshot from 2021-10-05 16-56-01](https://user-images.githubusercontent.com/81678524/135983264-4b0e0a81-e314-4ebd-b970-41a45e975723.png)

Previous apollo 5.0 commit with lgsvl 2021.3 and this apollo 5.0 commit with lgsvl 2021.2 are entirely different localization position result.
![Screenshot from 2021-10-05 16-30-08](https://user-images.githubusercontent.com/81678524/135979301-32c4a676-a126-4ae1-9da1-9d08ee043237.png)
I don't know where it is.

I test Shalun. and It works
![Screenshot from 2021-10-05 16-58-09](https://user-images.githubusercontent.com/81678524/135984772-bc5e06f8-de9b-4e2e-b7f4-afe29d5cbfb6.png)

I did not test highway101.
Borregas Ave always work well.

I always thank you for your help.
-Linsha

**Comment by EricBoiseLGSVL** [comment]
Glad to help.  Did you build the SF map for Apollo with the updated map? We tested this and found the correct rotation.  Are you still loading the old map in dreamview?

**Comment by diaokaitian** [comment]
has this issue solved？

**Comment by EricBoiseLGSVL** [comment]
be sure you are ignoring map origin.  We are sunsetting simulator and will no longer be pushing any updates, sorry.

**Comment by marcusvinicius178** [comment]
Hi @EricBoiseLGSVL could you please tell me where can I **find the translation/conversion of localization module from LGSVL and ApolloStack in the source code?**

Now I am using CARLA simulator and the localization is generated by ground truth.
We would like to use real data (GPS, IMU, odometry, Lidar) with CARLA-Apollo 5.0. 
However, the localization is already triggered within CARLA-Apollo ( does not need to activate localization module at control tab in Apollo, and if activate anything changes).

I did not found in files **bridge.sh** and **bootstrap_lgsvl.sh** how the "magic trick" is done. 

I mean Apollo localization module should perform all the algorithms calculation and return the localization do CARLA as it happens in LGSVL. But how did you implement this translation and everything regarding to localization LGSVL-CARLA?

Could you please provide me a direction, where should I start?

**Comment by EricBoiseLGSVL** [comment]
I suggest looking in the cyber bridge classes. I am not familiar with how carla is doing the conversion but we convert before a message is sent over bridge so Apollo can understand.

**Comment by marcusvinicius178** [comment]
Thanks a lot for the reply @EricBoiseLGSVL I am taking a look for a direction :)

 I forgot to communicate you about our last talk, I have solved that internal simulator issue...fixing the docker flags and Dockerfile: https://github.com/AuroAi/carla_apollo_bridge/issues/30

---------------------------------------------------------------------------------------------------------------------------
Now I must "build" of fix this Carla-Apollo bridge to work with data from localizations sensors instead ground truth: https://github.com/AuroAi/carla_apollo_bridge/issues/19

...Now about your hint: "take a look at lgsvl implementation of **cyber bridge classes**"... 
Could you say if I am looking to the correct files: 

-------------------------------------------------------------------------------------------------------------------------------
Localization translation is written in the Files from here?

https://github.com/lgsvl/apollo-5.0/tree/simulator/cyber/class_loader

or from here?

https://github.com/lgsvl/apollo-5.0/tree/simulator/cyber/class_loader/utility

or both or also in **other file**s?

-------------------------------------------------------------------------------------------------------------------------------

Could you please if not ask too much forward me the proper links (of lgsvl-apollo5 fork) related to this topic  (code where they are implemented)?

Thanks very much!

**Comment by EricBoiseLGSVL** [comment]
No, I don't think this is the location.  I think what you need to do is make conversion data in Carla and then create protobuf files that Apollo can understand.  Sorry I can't dig into this more.

**Comment by Addi2020** [comment]
@EricBoiseLGSVL 
I use Apollo7.0 and lgsvl 2021.3 version.

I find the same issue with SF map, The SF map is rotated in dream view and cannot adjust with the map in LGSVL simulator.  How to solve this issue?

**Comment by EricBoiseLGSVL** [comment]
@Addi2020 sorry we are no longer working on simulator.  Check #1994 for solutions.  I remember ignore map origin needed to be toggled but I don't have more than that, sorry.

**Comment by xiejingjacob** [comment]
@Addi2020 
Have you solve this problem?

I meet the some problem

**Comment by Addi2020** [comment]
@xiejingjacob sorry I am not working on it anymore. but I think this issue can be solved by uploading SF base_map.bin file to UNITY and rotating the whole map and then exporting the HD map again. But still bear in mind, that you may need the rotation points (x,y,z) from LGSVL SF map because you do not know much to rotate or you can just do it by the hit and trail rule.
